### PR TITLE
refactor(dev-infra): extract the parsing of a range of commits into a util function

### PR DESCRIPTION
**Pull request by josephperrott**
A utility function to parse a range of commits allows for other tooling to assess
a range of commits.

### CODE REVIEW

File: dev-infra/commit-message/validate.ts [hunk 1]
```diff
@@ -28,9 +28,15 @@ const COMMIT_BODY_URL_LINE_RE = /^https?:\/\/.*$/;
 
 /** Validate a commit message against using the local repo's config. */
 export function validateCommitMessage(
-    commitMsg: string, options: ValidateCommitMessageOptions = {}): ValidateCommitMessageResult {
+    commit: ParsedCommitMessage,
+    options?: ValidateCommitMessageOptions): ValidateCommitMessageResult;
+export function validateCommitMessage(
+    commitMsg: string, options?: ValidateCommitMessageOptions): ValidateCommitMessageResult;
```
Comment (gkalpak) [inline dev-infra/commit-message/validate.ts:34]: Is there any benefit in having these overloads here?

File: dev-infra/commit-message/utils.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * Custom git log format that provides the commit header and body, separated as expected with the
+   * custom separator as the trailing value.
+   */
+  const gitLogFormat = `%s%n%n%b${randomValueSeparator}`;
+
+  // Retrieve the commits in the provided range.
+  const result = exec(`git log --reverse --format=${gitLogFormat} ${range}`);
+  if (result.code) {
+    throw new Error(`Failed to get all commits in the range: \n  ${result.stderr}`);
```
Comment (gkalpak) [inline dev-infra/commit-message/utils.ts:26]: Nit:

```suggestion
    throw new Error(`Failed to get all commits in the range:\n  ${result.stderr}`);
```

File: dev-infra/commit-message/utils.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  return result
+      // Separate the commits from a single string into individual commits.
+      .split(randomValueSeparator)
+      // Remove extra space before and after each commit message.
+      .map(l => l.trim())
+      // Remove any superfluous lines which remane from the split.
```
Comment (gkalpak) [inline dev-infra/commit-message/utils.ts:34]:
```suggestion
      // Remove any superfluous lines which remained from the split.
```

File: dev-infra/commit-message/validate-range/validate-range.ts [hunk 4]
```diff
@@ -6,55 +6,39 @@
...
 
-  // Check each commit in the commit range.  Commits are allowed to be fixup commits for other
-  // commits in the provided commit range.
-  const allCommitsInRangeValid = commits.every((m, i) => {
+  /**
+   * Whether all commits in the range are valid.
+   *
+   * Check each commit in the commit range. Commits are allowed to be fixup commits for other
+   * commits in the provided commit range.
+   */
```
Comment (gkalpak) [inline dev-infra/commit-message/validate-range/validate-range.ts:33]: Nit: There doesn't seem to be any need for JSDoc-style block comments (`/** ... */`) inside this function.

**Review by gkalpak — APPROVED** [review]
A couple of nits. Otherwise lgtm :+1:

File: dev-infra/commit-message/validate-range/validate-range.ts (hunk 4 above)
Comment (josephperrott) [reply dev-infra/commit-message/validate-range/validate-range.ts:33]: Made it a bit more succinct.

**Pull request closed without merging**